evalml/tests/component_tests/test_arima_regressor.py

```diff
@@ -18,78 +20,191 @@ def test_problem_types():
+    else:
+        if dates_shape != 2:
+            X_, y_ = clf._format_dates(date_index, X, y, False)
+            assert X_.index.equals(y_.index)
+        elif dates_shape == 2:
+            with pytest.raises(ValueError, match='Found 2 columns'):
+                clf._format_dates(date_index, X, y, False)
+
+
 def test_feature_importance(ts_data):
     X, y = ts_data
     clf = ARIMARegressor()
     clf.fit(X, y)
     clf.feature_importance == np.zeros(1)
 
 
-def test_fit_predict_ts_with_X_and_y_index(ts_data):
-    X, y = ts_data
+def test_fit_predict_ts_with_datetime_in_X_column(ts_data_seasonal):
+    X, y = ts_data_seasonal
     assert isinstance(X.index, pd.DatetimeIndex)
     assert isinstance(y.index, pd.DatetimeIndex)
 
-    a_clf = arima.ARIMA(endog=y, exog=X, order=(1, 0, 0), trend='n', dates=X.index)
-    clf = a_clf.fit()
-    y_pred_a = clf.predict()
+    m_clf = ARIMARegressor(d=None)
+    m_clf.fit(X=X[:250], y=y[:250])
+    y_pred = m_clf.predict(X=X[250:])
 
-    m_clf = ARIMARegressor(p=1, d=0, q=0)
-    clf_ = m_clf.fit(X=X, y=y)
-    y_pred = clf_.predict(X=X)
+    X['Sample'] = pd.date_range(start='1/1/2016', periods=500)
 
-    assert (y_pred == y_pred_a).all()
+    dt_clf = ARIMARegressor(d=None)
+    dt_clf.fit(X=X[:250], y=y[:250])
+    y_pred_dt = dt_clf.predict(X=X[250:])
 
+    assert isinstance(y_pred_dt, woodwork.DataColumn)
+    pd.testing.assert_series_equal(y_pred.to_series(), y_pred_dt.to_series())
 
-def test_fit_predict_ts_with_X_not_y_index(ts_data):
-    X, y = ts_data
+
+def test_fit_predict_ts_with_only_datetime_column_in_X(ts_data_seasonal):
+    X, y = ts_data_seasonal
+    assert isinstance(X.index, pd.DatetimeIndex)
+    assert isinstance(y.index, pd.DatetimeIndex)
+
+    fh_ = forecasting.ForecastingHorizon(y[250:].index, is_relative=False)
+
+    a_clf = sktime_arima.AutoARIMA()
+    clf = a_clf.fit(y=y[:250])
+    y_pred_sk = clf.predict(fh=fh_)
+
+    X = X.drop(["features"], axis=1)
+
+    m_clf = ARIMARegressor(d=None)
+    m_clf.fit(X=X[:250], y=y[:250])
+    y_pred = m_clf.predict(X=X[250:])
+
+    assert (y_pred_sk.to_period('D') == y_pred.to_series()).all()
+
+
+def test_fit_predict_ts_with_X_and_y_index_out_of_sample(ts_data_seasonal):
+    X, y = ts_data_seasonal
     assert isinstance(X.index, pd.DatetimeIndex)
     assert isinstance(y.index, pd.DatetimeIndex)
 
-    a_clf = arima.ARIMA(endog=y, exog=X, order=(1, 0, 0), trend='n', dates=X.index)
-    clf = a_clf.fit()
-    y_pred_a = clf.predict()
+    fh_ = forecasting.ForecastingHorizon(y[250:].index, is_relative=False)
+
+    a_clf = sktime_arima.AutoARIMA()
+    clf = a_clf.fit(X=X[:250], y=y[:250])
+    y_pred_sk = clf.predict(fh=fh_, X=X[250:])
+
+    m_clf = ARIMARegressor(d=None)
+    m_clf.fit(X=X[:250], y=y[:250])
+    y_pred = m_clf.predict(X=X[250:])
+
+    assert (y_pred_sk.to_period('D') == y_pred.to_series()).all()
+
+
+def test_fit_predict_ts_with_X_and_y_index(ts_data_seasonal):
```

Comment:
with the index tests you can also mock out `_get_dates` and  do `mock_get_dates.assert_called_with()` to make sure the right index is passed through.